### ISSUE TITLE
Package export * behavior is incorrect

### ISSUE THREAD

**Issue by mfedderly** [rollup/rollup#1800]
Having an `export * from "package"` statement seems to cause the generated javascript to export things from the wrong source. I should mention that this code is getting transpiled from typescript, so I'd get an error if any of the exports had name collisions.

[Error Repro](https://rollupjs.org/repl?version=0.52.1&shareable=JTdCJTIybW9kdWxlcyUyMiUzQSU1QiU3QiUyMm5hbWUlMjIlM0ElMjJtYWluLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMmltcG9ydCUyMCU3QmZuJTdEJTIwZnJvbSUyMCU1QyUyMi4lMkZjaHVuazElNUMlMjIlM0IlNUNuJTVDbmZuKCU1QyUyMmZvbyU1QyUyMiklMjIlN0QlMkMlN0IlMjJuYW1lJTIyJTNBJTIyY2h1bmsxLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMmV4cG9ydCUyMColMjBmcm9tJTIwJTVDJTIycGtnJTVDJTIyJTNCJTVDbmV4cG9ydCUyMColMjBmcm9tJTIwJTVDJTIyLiUyRmNodW5rMiU1QyUyMiUzQiUyMiU3RCUyQyU3QiUyMm5hbWUlMjIlM0ElMjJjaHVuazIuanMlMjIlMkMlMjJjb2RlJTIyJTNBJTIyZXhwb3J0JTIwZnVuY3Rpb24lMjBmbigpJTIwJTdCY29uc29sZS5sb2coLi4uYXJndW1lbnRzKSUzQiU3RCUyMiU3RCU1RCUyQyUyMm9wdGlvbnMlMjIlM0ElN0IlMjJmb3JtYXQlMjIlM0ElMjJlcyUyMiUyQyUyMmdsb2JhbHMlMjIlM0ElN0IlN0QlMkMlMjJuYW1lJTIyJTNBJTIybXlCdW5kbGUlMjIlMkMlMjJhbWQlMjIlM0ElN0IlMjJpZCUyMiUzQSUyMiUyMiU3RCU3RCUyQyUyMmV4YW1wbGUlMjIlM0FudWxsJTdE)
[Similar but working](https://rollupjs.org/repl?version=0.52.1&shareable=JTdCJTIybW9kdWxlcyUyMiUzQSU1QiU3QiUyMm5hbWUlMjIlM0ElMjJtYWluLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMmltcG9ydCUyMCU3QmZuJTdEJTIwZnJvbSUyMCU1QyUyMi4lMkZjaHVuazElNUMlMjIlM0IlNUNuJTVDbmZuKCU1QyUyMmZvbyU1QyUyMiklMjIlN0QlMkMlN0IlMjJuYW1lJTIyJTNBJTIyY2h1bmsxLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMmV4cG9ydCUyMColMjBmcm9tJTIwJTVDJTIyLiUyRmNodW5rMiU1QyUyMiUzQiU1Q25leHBvcnQlMjAqJTIwZnJvbSUyMCU1QyUyMi4lMkZjaHVuazMlNUMlMjIlM0IlMjIlN0QlMkMlN0IlMjJuYW1lJTIyJTNBJTIyY2h1bmsyLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMmV4cG9ydCUyMGZ1bmN0aW9uJTIwZm4oKSUyMCU3QmNvbnNvbGUubG9nKC4uLmFyZ3VtZW50cyklM0IlN0QlMjIlN0QlMkMlN0IlMjJuYW1lJTIyJTNBJTIyY2h1bmszLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMmV4cG9ydCUyMGZ1bmN0aW9uJTIwZm4yKCklMjAlN0Jjb25zb2xlLmxvZyguLi5hcmd1bWVudHMpJTNCJTdEJTIyJTdEJTVEJTJDJTIyb3B0aW9ucyUyMiUzQSU3QiUyMmZvcm1hdCUyMiUzQSUyMmVzJTIyJTJDJTIyZ2xvYmFscyUyMiUzQSU3QiU3RCUyQyUyMm5hbWUlMjIlM0ElMjJteUJ1bmRsZSUyMiUyQyUyMmFtZCUyMiUzQSU3QiUyMmlkJTIyJTNBJTIyJTIyJTdEJTdEJTJDJTIyZXhhbXBsZSUyMiUzQW51bGwlN0Q=)

It seems to me that exporting an entire package is a bad idea anyhow, so I I'm going to go ahead and remove it. I figured I would file a bug anyhow in case there's a better way of handling this situation. I'm unsure of the module semantics around exports, does ordering matter for multiple `export *` statements?

**Comment by guybedford** [comment]
Unfortunately this is a consequence of the way Rollup has to assume external star exports might catch any given export name. So ordering does matter here yes.

We could possibly provide options to configure this sort of thing but I get the feeling that tweaking the ordering in the code base is an easier more sensible fix, even if this is a somewhat Rollup-specific scenario.

**Comment by guybedford** [comment]
Closing, but happy to continue discussion on this.

**Comment by mfedderly** [comment]
I have fixed the issue by changing all of my exports to be explicitly named. Any chance to at least get a warning when this scenario happens? It would've saved me having to dig through the generated code to figure out what caused the issue.

**Comment by guybedford** [comment]
Reopening to note this should be a warning.

**Comment by shellscape** [comment]
@lukastaegert is this still an issue in the latest version?

**Comment by lukastaegert** [comment]
Yes, it is still an issue. Just to recapitulate for future reference, the issue is that in any situation where an external star re-export overlaps with another star import (external or not), we want to display a warning that the behaviour might not be 100% accurate as Rollup will
* prioritize internal reexports over external ones
* not throw errors when there is a name conflict (which the spec says should happen as far as I know)
